python/ray/serve/controller.py

```diff
@@ -407,10 +405,28 @@ def deploy_app(
         if update_time:
             self.deployment_timestamp = time.time()
 
+        deployment_override_options = config.dict(
```

Comment:
The schema no longer uses aliases, so we don't need to create a separate `dict` where `by_alias` is `True`. We can use `config_dict.get("deployments", [])` here.